### ISSUE TITLE
S3: NIRSpec full dataset causes background flagging procedure to assume all pixels are bad. 

### ISSUE THREAD

**Issue by AarynnCarter** [kevin218/Eureka#193]
When moving from the Tiny dataset to the full dataset for NIRSpec, the extra frames introduce more NaN's per pixel over the full timespan (which get converted to 0's) which in turn causes the background flagging procedure to think every pixel is bad. 

The solution for me was to duplicate the NIRCam function currently being used into the NIRSpec flag_bg function and adjust the following two lines, so that they mask out any values that are equal to 0.  

```
    bgerr1  = np.ma.median(np.ma.masked_equal(data.suberr[:,  :y1],0))
    bgerr2  = np.ma.median(np.ma.masked_equal(data.suberr[:,y2:  ],0))
```

Bigger questions underpin this...

1) Do we still need to be converting the NaN's to zeroes? 
2) Why are there so many NaN's in the NIRSpec simulated data
3) Will this be an issue for the real data? I.e. do we need to spend a lot of time on this

**Comment by kevin218** [comment]
With the switch to masked arrays, it doesn't make sense to convert bad pixels to 0.

**Comment by AarynnCarter** [comment]
Okay unfortunately this wasn't as quick of a fix as removing the lines converting the NaN's to zeroes. If I do so the MAD plot is empty. Interestingly, the BGSUB images look quite different between the adjusted flag_bg I described above, and the default NIRCam flag_bg function. With the default method, it looks like the background is barely even subtracted, and the background pixel flux changes a lot between integrations (same scaling), for the adjusted method it looks much more uniform. 

Will take a look later this week, but wanted to provide a quick update. 

![fig3101-2D_LC](https://user-images.githubusercontent.com/23636747/160507779-ee2b967d-d020-4c49-b6de-3b27adbe0ecb.png)

**Comment by AarynnCarter** [comment]
On a separate point I also think this issue may be what drove the strict requirements on bg_hw that @erinmmay metioned in #121, as a smaller bg_hw could provide some pixels where there are no NaN's present throughout the time series (i.e. where the spectrum is).

**Comment by AarynnCarter** [comment]
Okay, I've figured it out, and this is an issue with the NIRSpec data itself. Whilst the NaN's are being correctly masked in the SCI data array, the ERR array is filled with zeroes. I've looked back at data I've made going back to May 2021 and they all exhibit a similar abundance of zeroes. 

For now, I'm thinking we keep the adjustments to `bgerr1` and `bgerr2` I've listed above. I expect this level of zeroes won't be the case for the real data, and I'm not sure I have the time to hand to dig into this further. Especially considering we're getting reasonable results (as has Zafar/David with a different pipeline). 

We just have to remember to "undo" this masking of zeroes for the real data. 

![image](https://user-images.githubusercontent.com/23636747/161343093-6b1491c3-0cc8-42b5-b9ed-5844bf5f14ee.png)
